Runtime/Interaction/SnapZone.cs

```diff
@@ -182,10 +185,28 @@ protected override void Awake()
             }
         }
 
+        internal void AddHoveredInteractable(XRBaseInteractable target)
+        {  
+            if (target != null)
+            {
+                hoverTargets.Add(target);
+            }
+        }
+
+        internal void RemoveHoveredInteractable(XRBaseInteractable target)
+        {
+            if (target != null)
+            {
+                hoverTargets.Remove(target);
+            }
+        }
```

Comment:
Can you validate if the items exist in the list before doing any operation?

Reply:
List.Remove does that for you. It goes through the list until it finds the element, and then either returns false or calls RemoveAt() and returns true.